opensrp-giz-malawi/src/main/java/org/smartregister/giz/application/GizMalawiApplication.java

```diff
@@ -363,6 +366,22 @@ public void onCreate() {
 
         initMinimumDateForReportGeneration();
 
+        updateBaseUrlFromBuildConfigChange();
+    }
+
+    private void updateBaseUrlFromBuildConfigChange() {
```

Comment:
Could we have tests(s) for this funtion

Reply:
Definitely. However, I'm having trouble running tests and it's not clear what the issue is.

```
The following annotation processors are not incremental: jetified-butterknife-7.0.0.jar (com.jakewharton:butterknife:7.0.0), jetified-eventbus-annotation-processor-3.1.1.jar (org.greenrobot:eventbus-annotation-processor:3.1.1).
Make sure all annotation processors are incremental to improve your build speed.
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
> Task :opensrp-giz-malawi:packageDebugUnitTestForUnitTest
> Task :opensrp-giz-malawi:bundleDebugClasses
> Task :opensrp-giz-malawi:compileDebugUnitTestJavaWithJavac
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
> Task :opensrp-giz-malawi:generateDebugUnitTestConfig
> Task :opensrp-giz-malawi:testDebugUnitTest
java.lang.NoClassDefFoundError: jdk/internal/reflect/GeneratedSerializationConstructorAccessor1
	at jdk.internal.reflect.GeneratedSerializationConstructorAccessor1.newInstance(Unknown Source)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
	at java.base/java.io.ObjectStreamClass.newInstance(ObjectStreamClass.java:1092)
	at java.base/java.io.ObjectInputStream.readOrdinaryObject(ObjectInputStream.java:2180)
	at java.base/java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1679)
	at java.base/java.io.ObjectInputStream.readObject(ObjectInputStream.java:493)
	at java.base/java.io.ObjectInputStream.readObject(ObjectInputStream.java:451)
	at org.gradle.process.internal.worker.child.SystemApplicationClassLoaderWorker.call(SystemApplicationClassLoaderWorker.java:122)
	at org.gradle.process.internal.worker.child.SystemApplicationClassLoaderWorker.call(SystemApplicationClassLoaderWorker.java:69)
	at worker.org.gradle.process.internal.worker.GradleWorkerMain.run(GradleWorkerMain.java:68)
	at worker.org.gradle.process.internal.worker.GradleWorkerMain.main(GradleWorkerMain.java:73)
Caused by: java.lang.ClassNotFoundException: jdk.internal.reflect.GeneratedSerializationConstructorAccessor1
	at java.base/java.net.URLClassLoader.findClass(URLClassLoader.java:476)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:589)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
	... 11 more
> Task :opensrp-giz-malawi:testDebugUnitTest FAILED
> Task :buildDashboard UP-TO-DATE
```

I work on these tests and the issue once done generating an APK. This will be tracked on this issue https://github.com/opensrp/support/issues/273